### DISCUSSION TITLE
Windows V6.0 test build

### DISCUSSION THREAD

**Discussion by gusmanb** [gusmanb/logicanalyzer#148] [General]
For Windows users that would like to test the new V6.0 and be part of the debugging, [here is the current build](https://mega.nz/file/2Cw3CCBQ#3Xywe3JrMpgKh-qGcCSVj7nYNpo6ZPSRh1Nx4tLf5Vc).

Remember to use one of the [new firmwares](https://mega.nz/file/3SZCDQzD#krl3m4G7HKwNuCi6rsmw_yeyi9CK_usz6gGNyXah-uY).

Python 3.x must be installed on the system, minimum version 3.7. If the analyzer is not able to locate the version you can add a "python.cfg" file in the app folder containing the path to the "libpython3.x.so".

If you find bugs or errors please notify them here.

## Keep in mind!

This is a test build and to ease things I will upload multiple versions with updates answering to the posts with bugs, please read the thread to find which version is the latest one.

Cheers.

**Comment by tingox** [comment]
looks like mega decided to invalidate these links too.

**Reply by gusmanb** [reply]
Here is the link to the folder with all the current versions: https://mega.nz/folder/SGxDHAZL#afLGgQbJAaqOYXjhJBwokQ

**Comment by MichaelM4** [comment]
Version 5.1 was nice.  Version 6.0 is very nice.
No bugs found yet.

**Comment by MichaelM4** [comment]
I know beggars should not be choosy, but :)
Any news on version 6?

**Reply by JimDrewGH** [reply]
You can download V6 from the link above.

**Reply by MichaelM4** [reply]
That version is dated 10/18.  Is there a newer version?

**Reply by JimDrewGH** [reply]
Not that I am aware of - that is the version I have been using with the new board design.

**Reply by sstvmaster** [reply]
Problems finding python313.dll.

My system is Windows 10, x64
- python 3.13 x64, is installed in C:\Users\xxx\AppData\Local\Programs\Python\Python313\python313.dll
- this entry is in python.cfg, located in LogicAnalyzer folder

The LogicAnalyzer software gives me always the error message that python is not installed corectly.
If i open the command line prompt an type "py" python shows his information

`Python 3.13.1 (tags/v3.13.1:0671451, Dec  3 2024, 19:06:28) [MSC v.1942 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.`

here is the PythonInitLog.txt:
`12.12.2024 22:31 -> Initializing python system...
12.12.2024 22:31 -> Reading python path from config file...
12.12.2024 22:31 -> Stablished path: C:\Users\xxx\AppData\Local\Programs\Python\Python313\python313.dll

12.12.2024 22:31 -> Initializing decoders...
12.12.2024 22:31 -> Error initializing python engine: The type initializer for 'Delegates' threw an exception. -    at Python.Runtime.Runtime.Delegates.get_Py_IsInitialized()
   at Python.Runtime.Runtime.Py_IsInitialized()
   at Python.Runtime.Runtime.<>c.<Initialize>b__32_0()
   at Python.Runtime.Runtime.TryUsingDll[T](Func`1 op)
   at Python.Runtime.Runtime.Initialize(Boolean initSigs)
   at Python.Runtime.PythonEngine.Initialize(IEnumerable`1 args, Boolean setSysArgv, Boolean initSigs)
   at Python.Runtime.PythonEngine.Initialize(Boolean setSysArgv, Boolean initSigs)
   at Python.Runtime.PythonEngine.Initialize()
   at SigrokDecoderBridge.SigrokPythonEngine.EnsureInitialized() in F:\LogicAnalyzerRepo\Software\LogicAnalyzer\LogicAnalyzer\SigrokDecoderBridge\SigrokPythonEngine.cs:line 99`

i also tested with py 3.12 x64

Many thanks

**Reply by gusmanb** [reply]
https://github.com/gusmanb/logicanalyzer/discussions/153

**Reply by sstvmaster** [reply]
Now it works with py 3.12, i also had to delete python.cfg.

many thanks

**Reply by mz1874** [reply]
Have you fixed the error , as i meeted it in python 3.13. I even tried used a totally new computer , the log still is 
at SigrokDecoderBridge.SigrokPythonEngine.EnsureInitialized() in F:\LogicAnalyzerRepo\Software\LogicAnalyzer\LogicAnalyzer\SigrokDecoderBridge\SigrokPythonEngine.cs:line 99

**Reply by mz1874** [reply]
> Have you fixed the error , as i meeted it in python 3.13. I even tried used a totally new computer , the log still is at SigrokDecoderBridge.SigrokPythonEngine.EnsureInitialized() in F:\LogicAnalyzerRepo\Software\LogicAnalyzer\LogicAnalyzer\SigrokDecoderBridge\SigrokPythonEngine.cs:line 99

I try to install 3.12

**Comment by billgrundmann** [comment]
Fiirst - thank you for the excellent work and software.

I have just installed the latest V6 firmware and software but now have the python issue for loading the decoders.  After tying several python versions, I have found that python-3.14.2-amd64 does not yet work, but python-3.13.11-amd64 does -- at least on my system, Windows 11.

**Reply by gusmanb** [reply]
Yes, this is known, Python.net only supports up to Python V3.13. I have updated the docs to reflect it, when I last updated the docs the last version available was Python 3.13.
